### PR TITLE
chore: use sqids instead of sha256 hashing

### PR DESCRIPTION
**Pull request by juliusmarminge**
Returns a key like `KJNGYKlZPq92Lq0BVqOBeV4QXKqyYlocntI7CxzdHiOfLwFm` instead of a 172ch long base64 string

### CODE REVIEW

File: packages/shared/src/crypto.ts [hunk 1]
```diff
@@ -91,22 +105,27 @@ export const generateKey = (
...
+    const encodedFileSeed = new SQIds({ alphabet, minLength: 36 }).encode([
+      Math.abs(Hash.string(hashParts)),
+    ]);
+    const encodedApiKey = new SQIds({ alphabet, minLength: 12 }).encode([
+      Math.abs(Hash.string(apiKey)),
+    ]);
 
     // Concatenate them and encode as base64
-    return Encoding.encodeBase64([encodedApiKey, encodedFileSeed].join(":"));
+    return [encodedApiKey, encodedFileSeed].join(KEY_SEPARATOR);
```
Comment (juliusmarminge) [inline packages/shared/src/crypto.ts:117]: Is this URL safe? I chose it cause it "looks" like a normal character so it doesn't stand out as much xd

could also base64 encode the key, but it does add to it's length

```sh
KJNGYKlZPq92|4wGwXUsC5WekfPrntiTE0qUyIo91O87zXwNg => 49ch
S0pOR1lLbFpQcTkyfDR3R3dYVXNDNVdla2ZQcm50aVRFMHFVeUlvOTFPODd6WHdOZw== 68ch
```

Comment (markflorkowski) [reply packages/shared/src/crypto.ts:117]: Pipe is not urlsafe, could use dash?

Comment (markflorkowski) [reply packages/shared/src/crypto.ts:117]: do they need to be separated?

Comment (juliusmarminge) [reply packages/shared/src/crypto.ts:117]: i guess not really if we do a `startsWith` check instead.

I think the reason for the separation initially was adding a timing safe equals but that might be overkill?

Comment (markflorkowski) [reply packages/shared/src/crypto.ts:117]: Yeah, I think it's fine if it isn't timing safe

**Pull request merged**